### ISSUE TITLE
Markdown writer: flag for generating inline rather than reference-style footnotes

### ISSUE THREAD

**Issue by ltrgoddard** [jgm/pandoc#2567]
When converting to Markdown, there is currently no way to make the writer use Pandoc's inline footnotes ("this is some text^[this is a footnote]") in generated text, rather than reference-style notes at the end of the document (see [this](http://stackoverflow.com/questions/33984689/generate-inline-rather-than-list-style-footnotes-in-pandoc-markdown-output) question on StackOverflow). A flag to specify this behaviour  would be useful for anyone converting from, say, a relatively long and heavily noted Docx, who then also needed to add new footnotes or modify the order of existing ones in the resuting Markdown (no more numbering conflicts, etc.) - not as niche a use-case as it sounds, I'd wager.

Just a thought!

**Comment by frederik-elwert** [comment]
Yes, I stumbled upon this several times before. I would really appreciate a flag like this.

Btw, thanks for the script, @ltrgoddard! Very helpful!

**Comment by ltrgoddard** [comment]
No problem!

**Comment by EvanEdwards** [comment]
@ltrgoddard  Thanks for the script!  It would be better as a flag, but you saved me some time.

**Comment by ltrgoddard** [comment]
No problem!

**Comment by jgm** [comment]
The problem is that inline notes are only possible when the note doesn't contain block-level content.  So you can't really have an option to use only inline notes.  The best we could do would be to use inline notes when possible.  I think that's a bit ugly and not simple conceptually, so I'd prefer not to add such a flag.

Note that we now provide an option that will let you have your notes at the end of the enclosing block (e.g. paragraph) or the end of the section rather than the end of the document.  This should help for at least some of the cases where you wanted inline notes.
